### PR TITLE
[risk=low][RW-10991] Runtime conf panel: add types to props, and reorganize components

### PR DESCRIPTION
**Pull request by jmthibault79**
Many components used by Runtime Configuration Panel had untyped props.  I added them.

I also moved some configuration panel components common to GCE Runtimes and GKE Apps to a new folder `common-env-conf-panels` - naming suggestions welcome.

No functional changes intended or expected.  Tested by running locally.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/gke-app-configuration-panels/create-gke-app.tsx [hunk 1]
```diff
@@ -94,12 +96,31 @@ export const CreateGkeApp = ({
 
   const persistentDiskRequest: PersistentDiskRequest =
     disk ?? defaultConfig.persistentDiskRequest;
+  const { kubernetesRuntimeConfig } = defaultConfig;
+  const machine: Machine = findMachineByName(
+    kubernetesRuntimeConfig.machineType
+  );
+  const analysisConfig: AnalysisConfig = {
```
Comment (jmthibault79) [inline ui/src/app/components/gke-app-configuration-panels/create-gke-app.tsx:103]: this is a modified version of the original `createAppRequestToAnalysisConfig()`

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 2]
```diff
@@ -106,8 +120,8 @@ const CreatePanel = ({
           analysisConfig,
           status,
         }}
-        onPause={() => setRuntimeStatus(RuntimeStatusRequest.Stop)}
-        onResume={() => setRuntimeStatus(RuntimeStatusRequest.Start)}
+        onPause={async () => setRuntimeStatus(RuntimeStatusRequest.Stop)}
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:123]: onPause/onResume now have types which return Promises so they must be async

File: ui/src/app/utils/cdr-versions.tsx [hunk 3]
```diff
@@ -39,9 +41,12 @@ function hasDefaultCdrVersion(
 // does not consider tier; IDs are globally unique, enforced by the API DB
 function findCdrVersion(
   cdrVersionId: string,
-  cdrTiers: CdrVersionTiersResponse
+  cdrTiers: CdrVersionTiersResponse | CdrVersionStore
```
Comment (jmthibault79) [inline ui/src/app/utils/cdr-versions.tsx:44]: these types only differ in that `tiers` may be optional in one or the other.  Adding types clarified that we were calling this function with both.

File: ui/src/app/utils/index.tsx [hunk 4]
```diff
@@ -275,6 +281,9 @@ export const connectReplaySubject = <T extends {}>(
 export const withCurrentWorkspace = () => {
   return connectBehaviorSubject(currentWorkspaceStore, 'workspace');
 };
+export interface WithCurrentWorkspace {
```
Comment (jmthibault79) [inline ui/src/app/utils/index.tsx:284]: matches the above

File: ui/src/app/utils/index.tsx [hunk 5]
```diff
@@ -320,6 +329,9 @@ export const withUserProfile = () => {
 export const withCdrVersions = () => {
   return withStore(cdrVersionStore, 'cdrVersionTiersResponse');
 };
+export interface WithCdrVersions {
```
Comment (jmthibault79) [inline ui/src/app/utils/index.tsx:332]: matches the above

File: ui/src/app/utils/stores.tsx [hunk 6]
```diff
@@ -287,10 +287,12 @@ export function useStore<T>(theStore: Atom<T>) {
  * HOC that injects the value of the given store as a prop. When the store changes, the wrapped
  * component will re-render
  */
-export const withStore = (theStore, name) => (WrappedComponent) => {
-  return (props) => {
-    const value = useStore(theStore);
-    const storeProp = { [name]: value };
-    return <WrappedComponent {...props} {...storeProp} />;
+export const withStore =
+  <T,>(theStore: Atom<T>, name: string) =>
```
Comment (jmthibault79) [inline ui/src/app/utils/stores.tsx:291]: this funny syntax `<T,>` is how generics are represented in typescript

**Review by Qi77Qi — APPROVED** [review]
seems mostly the change is ascribing types and a bit of formatting diff 👍

File: ui/src/app/components/apps-panel/utils.tsx [hunk 7]
```diff
@@ -101,23 +100,6 @@ export const defaultSASConfig: CreateAppRequest = {
   },
 };
 
-export const createAppRequestToAnalysisConfig = (
-  createAppRequest: CreateAppRequest
-): Partial<AnalysisConfig> => {
```
[the remark this answers is not in the document]
Comment (Peter-Lavigne) [reply ui/src/app/components/apps-panel/utils.tsx:106]: Great idea 👍

**Review by Peter-Lavigne — APPROVED** [review]
Thank you! This makes the code _enormously_ more understandable for me when reading through it.

**Pull request merged**